okonomiyaki/file_formats/runtime.py

```diff
@@ -0,0 +1,201 @@
+        if metadata_version != "1.0":
+            msg = "Unsupported metadata version: {0!r}"
+            raise UnsupportedMetadata(msg.format(metadata_version))
+
+        return cls.from_json_dict(metadata_dict)
+
+    @classmethod
+    def from_json_dict(cls, data):
+        args = cls._from_json_dict(data)
+        return cls(*args)
+
+    @classmethod
+    def _from_json_dict(cls, data):
+        language = data["language"]
+        implementation = data["implementation"]
+        version = RuntimeVersion.from_string(data["version"])
+        platform = EPDPlatform.from_epd_string(data["platform"]).platform
+
+        build_revision = data["build_revision"]
+
+        executable = data["executable"]
+        paths = tuple(data["paths"])
+        post_install = tuple(data["post_install"])
+
+        return (
+            language, implementation, version, platform, build_revision,
+            executable, paths, post_install
+        )
+
+    @property
+    def filename(self):
+        disp_platform = EPDPlatform(self.platform).platform_name
+        disp_platform += "_" + str(self.platform.arch)
+        template = "{0.language}-{0.implementation}-{0.version}-{1}"
+        return template.format(self, disp_platform) + ".runtime"
+
+
+class JuliaRuntimeMetadataV1(RuntimeMetadataV1):
+    """ Class representing the metadata of a julia runtime package.
+    """
+
+
+@attributes
+class PythonRuntimeMetadataV1(RuntimeMetadataV1):
+    """ Class representing the metadata of a python runtime package.
+    """
+    scriptsdir = attr(validator=instance_of(six.text_type))
+    site_packages = attr(validator=instance_of(six.text_type))
+
+    @classmethod
+    def _from_json_dict(cls, data):
+        args = super(PythonRuntimeMetadataV1, cls)._from_json_dict(data)
+        scriptsdir = data["scriptsdir"]
+        site_packages = data["site_packages"]
+        return args + (scriptsdir, site_packages)
+
+
+_METADATA_KLASS_FACTORY = {
+    (MetadataVersion.from_string("1.0"), "python"): PythonRuntimeMetadataV1,
+    (MetadataVersion.from_string("1.0"), "julia"): JuliaRuntimeMetadataV1,
+}
+
+
+def runtime_metadata_factory(path_or_file):
+    """ Creates metadata object of the appropriate class from the given path.
+
+    Parameters
+    ----------
+    path_or_file: str or ZipFile
+        The path to the runtime package. May be a ZipFile instance.
+    """
+    def _factory_key_from_metadata(json_dict):
+        return (
+            MetadataVersion.from_string(json_dict["metadata_version"]),
+            json_dict["language"]
+        )
+
+    if isinstance(path_or_file, six.string_types):
+        with zipfile2.ZipFile(path_or_file) as zp:
+            metadata = _read_runtime_metadata_json(zp)
+    else:
+        metadata = _read_runtime_metadata_json(path_or_file)
+
+    json_dict = json.loads(metadata)
+    key = _factory_key_from_metadata(json_dict)
+    klass = _METADATA_KLASS_FACTORY.get(key)
```

Comment:
I assume that at a later version the class registry will be an optional argument correct?